plugins/pastefromword/filter/default.js

```diff
@@ -1541,34 +1541,140 @@
 		 */
 		extractFromRtf: function( rtfContent ) {
 			var ret = [],
-				rePictureHeader = /\{\\pict[\s\S]+?\\bliptag\-?\d+(\\blipupi\-?\d+)?(\{\\\*\\blipuid\s?[\da-fA-F]+)?[\s\}]*?/,
+				rePictureHeader = /(?:(?:\\\*)?\{\\(?:non)?shppict\s*)?\{\\pict[\s\S]+?\\bliptag\-?\d+(\\blipupi\-?\d+)?(\{\\\*\\blipuid\s?[\da-fA-F]+)?[\s\}]*?/,
 				rePicture = new RegExp( '(?:(' + rePictureHeader.source + '))([\\da-fA-F\\s]+)\\}', 'g' ),
 				wholeImages,
 				imageType;
 
+			rtfContent = removeHeadersAndFooters( rtfContent );
 			wholeImages = rtfContent.match( rePicture );
+
 			if ( !wholeImages ) {
 				return ret;
 			}
 
 			for ( var i = 0; i < wholeImages.length; i++ ) {
-				if ( rePictureHeader.test( wholeImages[ i ] ) ) {
-					if ( wholeImages[ i ].indexOf( '\\pngblip' ) !== -1 ) {
+				var currentImage = wholeImages[ i ];
+
+				if ( rePictureHeader.test( currentImage ) ) {
+					var id = getImageId( currentImage ),
+						imageDataIndex = CKEDITOR.tools.array.indexOf( ret, function( image ) {
+							return image.id === id;
+						} ),
+						isAlreadyExtracted = imageDataIndex !== -1 && ret[ imageDataIndex ].hex,
+						isWordArtShape = currentImage.indexOf( '\\defshp' ) !== -1,
+						isNonWordImage = currentImage.indexOf( '\\nonshppict' ) !== -1;
+
+					// This image is already extracted, it's WordArt shape or it's a non-Word version.
+					if ( isAlreadyExtracted || isWordArtShape || isNonWordImage ) {
+						continue;
+					}
+
+					if ( currentImage.indexOf( '\\pngblip' ) !== -1 ) {
 						imageType = 'image/png';
-					} else if ( wholeImages[ i ].indexOf( '\\jpegblip' ) !== -1 ) {
+					} else if ( currentImage.indexOf( '\\jpegblip' ) !== -1 ) {
 						imageType = 'image/jpeg';
 					} else {
-						continue;
+						imageType = 'unknown';
 					}
 
-					ret.push( {
-						hex: imageType ? wholeImages[ i ].replace( rePictureHeader, '' ).replace( /[^\da-fA-F]/g, '' ) : null,
+					var newImageData = {
+						id: id,
+						hex: imageType !== 'unknown' ?
+							currentImage.replace( rePictureHeader, '' ).replace( /[^\da-fA-F]/g, '' ) : null,
 						type: imageType
-					} );
+					};
+
+					if ( imageDataIndex !== -1 ) {
+						ret.splice( imageDataIndex, 1, newImageData );
+					} else {
+						ret.push( newImageData );
+					}
 				}
 			}
 
 			return ret;
+
+			function getImageId( image ) {
+				var blipUidRegex = /\\blipuid (\w+)\}/,
+					blipTagRegex = /\\bliptag(-?\d+)/,
+					blipUidMatch = image.match( blipUidRegex ),
+					blipTagMatch = image.match( blipTagRegex );
+
+				if ( blipUidMatch ) {
+					return blipUidMatch[ 1 ];
+				} else if ( blipTagMatch ) {
+					return blipTagMatch[ 1 ];
+				}
+
+				return null;
+			}
+
+			function removeHeadersAndFooters( rtfContent ) {
```

Comment:
I think this could be generalized to `removeGroups()` method and used like:

```javascript
removeGroups( rtfContent, /\{\\(?:header|footer)[lrf]?/ ); // passing content and regex or
removeGroups( rtfContent, [ 'header', 'footer' ] ); // simply passing group names
```